### DISCUSSION TITLE
V7 fieldArray typing

### DISCUSSION THREAD

**Discussion by 0xsven** [react-hook-form/react-hook-form#7848] [Q&A] [answered]
I have a controlled component with a field array. It receives a dynamic name attribute via props. It works fine, but the types are wrong. Can you help me to type it? I can't get it to work.

Sandbox: https://codesandbox.io/s/react-form-hooks-v7-fieldarray-z6ofmz?file=/src/inputs/FieldArrayInput.tsx

Screenshot of the problem:
![Screenshot 2022-02-18 at 15 33 11](https://user-images.githubusercontent.com/973379/154702262-2a51d081-6c2a-4ecf-941b-f602a2b9b9ab.png)

**Comment by Moshyfawn** [comment, marked as answer]
I'd say, it's a current RHF limitation which is gonna be addressed in the next major V8 release. See #7354
